clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java

```diff
@@ -306,7 +305,7 @@ public boolean hasExpired() {
     }
 
     synchronized List<KafkaMetric> metrics() {
```

Comment:
If we are going to change it then we should probably also add a comment for the method saying `// visible for testing`

Reply:
Done.  Thank you.  Please review and kick off test.

Reply:
This method is not only used in tests, so the comment seems wrong to me. Can we fix it?